libtenzir/builtins/connectors/tcp.cpp

```diff
@@ -385,49 +548,60 @@ class plugin final : public virtual loader_plugin<loader> {
 public:
   auto parse_loader(parser_interface& p) const
     -> std::unique_ptr<plugin_loader> override {
+    auto args = parse_args<loader_args>(p);
+    return std::make_unique<loader>(std::move(args));
+  }
+
+  auto parse_saver(parser_interface& p) const
+    -> std::unique_ptr<plugin_saver> override {
+    auto args = parse_args<saver_args>(p);
+    return std::make_unique<saver>(std::move(args));
+  }
+
+  template <class Args>
+  auto parse_args(parser_interface& p) const -> Args {
     auto parser = argument_parser{
       name(),
       fmt::format("https://docs.tenzir.com/docs/connectors/{}", name())};
-    auto uri = std::string{};
-    auto certfile = std::string{};
-    auto keyfile = std::string{};
-    auto tls = bool{};
-    auto listen = bool{};
-    auto keep_listening = bool{};
+    auto args = Args{};
+    auto uri = located<std::string>{};
     parser.add(uri, "<endpoint>");
-    parser.add("--tls", tls);
-    parser.add("-l,--listen", listen);
-    parser.add("-k,--keep-listening", keep_listening);
-    parser.add("--certfile", certfile, "TLS certificate");
-    parser.add("--keyfile", keyfile, "TLS private key");
+    if constexpr (std::is_same_v<Args, loader_args>) {
+      parser.add("-c,--connect", args.connect);
+    } else if constexpr (std::is_same_v<Args, saver_args>) {
+      parser.add("-l,--listen", args.listen);
+    }
+    parser.add("-o,--once", args.once);
```

Comment:
You should probably add a check that `--once` is only used in combination with `--listen`, since it's not obvious from the name anymore.

Reply:
I previously suggested calling the option `--listen-once` and having it imply `--listen`. Do you think that would be more intuitive?

Reply:
Yes, that sounds better

Reply:
We fixed the issue. We can reconsider the naming at a later point.

Reply:
Since it already caused confusion for at least two people, i.e.. Benno and myself, why not rename it now?

Reply:
Okay, I'll go with `--listen-once`.

Reply:
See e65814f.